src/comestible_inv.cpp

```diff
@@ -0,0 +1,2477 @@
+            print_color = thiscolor;
+            if( it_amt > 9999 ) {
+                it_amt = 9999;
+                print_color = selected ? hilite( c_red ) : c_red;
+            }
+            mvwprintz( window, 6 + x, amt_startpos, print_color, "%4d", it_amt );
+        }
+
+        //print weight column
+        double it_weight = convert_weight( sitem.weight );
+        size_t w_precision;
+        print_color = it_weight > 0 ? thiscolor : thiscolordark;
+
+        if( it_weight >= 1000.0 ) {
+            if( it_weight >= 10000.0 ) {
+                print_color = selected ? hilite( c_red ) : c_red;
+                it_weight = 9999.0;
+            }
+            w_precision = 0;
+        } else if( it_weight >= 100.0 ) {
+            w_precision = 1;
+        } else {
+            w_precision = 2;
+        }
+        mvwprintz( window, 6 + x, weight_startpos, print_color, "%5.*f", w_precision, it_weight );
+
+        //print volume column
+        bool it_vol_truncated = false;
+        double it_vol_value = 0.0;
+        std::string it_vol = format_volume( sitem.volume, 5, &it_vol_truncated, &it_vol_value );
+        if( it_vol_truncated && it_vol_value > 0.0 ) {
+            print_color = selected ? hilite( c_red ) : c_red;
+        } else {
+            print_color = sitem.volume.value() > 0 ? thiscolor : thiscolordark;
+        }
+        mvwprintz( window, 6 + x, vol_startpos, print_color, it_vol );
+
+        if( sitem.autopickup ) {
+            mvwprintz( window, 6 + x, 1, magenta_background( it.color_in_inventory() ),
+                       compact ? it.tname().substr( 0, 1 ) : ">" );
+        }
+        //get_consumable_item
+        char const *string_format = "";
+        islot_comestible it_c = get_edible_comestible( p, it ); // .type->comestible;
+
+        //print "CALORIES" column
+        int it_cal = p.kcal_for( it );
+        string_format = set_string_params( print_color, it_cal, selected );
+        mvwprintz( window, 6 + x, cal_startpos, print_color, string_format, it_cal );
+
+        //print "QUENCH" column
+        int it_q = it_c.quench;
+        string_format = set_string_params( print_color, it_q, selected );
+        mvwprintz( window, 6 + x, quench_startpos, print_color, string_format, it_q );
+
+        //print "JOY" column
+        int it_joy = p.fun_for( it ).first;
+        string_format = set_string_params( print_color, it_joy, selected, it.has_flag( "MUSHY" ) );
+        mvwprintz( window, 6 + x, joy_startpos, print_color, string_format, it_joy );
+
+        //TODO: g->u.can_estimate_rot()
+        //print "SPOILS IN" column
+        std::string it_spoils_string;
+        if( it_c.spoils > 0_turns ) {
+            if( !it.rotten() ) {
+                it_spoils_string = get_time_left_rounded( p, it );
+            }
+        }
+        print_color = selected ? hilite( c_white ) : c_cyan;
+        mvwprintz( window, 6 + x, expires_startpos, print_color, "%s", it_spoils_string );
+
+        /*
```

Comment:
Please remove commented code once you are finished debugging and move PR out of Draft state.